src/java.base/share/classes/java/math/BigInteger.java

```diff
@@ -1273,13 +1273,14 @@ private static BigInteger valueOf(int[] val) {
          * with just the very first value.  Additional values will be created
          * on demand.
          */
-        powerCache = new BigInteger[Character.MAX_RADIX+1][];
+        BigInteger[][] powerCache = new BigInteger[Character.MAX_RADIX+1][];
```

Comment:
Using a different local name will avoid the shadowing and make it more obvious that the initialization is being deliberately being done before the assignment to a volatile.

Reply:
Fixed.